src/function/9pfs.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+
+#include "usbg/usbg.h"
+#include "usbg/usbg_internal.h"
+#include "usbg/function/9pfs.h"
+
+#include <malloc.h>
+#ifdef HAS_GADGET_SCHEMES
+#include <libconfig.h>
+#endif
+
+struct usbg_f_9pfs {
+	struct usbg_function func;
+};
+
+GENERIC_ALLOC_INST(p9fs, struct usbg_f_9pfs, func);
+
+GENERIC_FREE_INST(p9fs, struct usbg_f_9pfs, func);
+
+static int p9fs_set_attrs(struct usbg_function *f, void *f_attrs)
+{
+	const char *dev_name = *(const char **)f_attrs;
```

Comment:
Would it be worth checking if `f_attrs` is `NULL` before dereferencing it?

Reply:
Sure it will not hurt.

Reply:
This should actually not be necessary since we call immediately after the cast 

return dev_name && dev_name[0] ? USBG_ERROR_INVALID_PARAM : USBG_SUCCESS;

Which will exactly do return NULL if f_attrs was NULL. So there is nothing left to be done.